agent/httpserver/listen.go

```diff
@@ -10,39 +11,171 @@ import (
 	"github.com/darvaza-proxy/darvaza/shared/net/bind"
 )
 
+// ServerListeners is the list of all listeners on a Server
+type ServerListeners struct {
+	Insecure []*net.TCPListener
+	Secure   []*net.TCPListener
+	Quic     []*net.UDPConn
+}
+
+// Close closes all listeners. Errors are ignored
+func (sl *ServerListeners) Close() error {
+	closeAll(sl.Insecure)
+	closeAll(sl.Secure)
+	closeAll(sl.Quic)
+	return nil
+}
+
 func closeAll[T io.Closer](s []T) {
 	for _, l := range s {
 		_ = l.Close()
 	}
 }
 
+func (*ServerListeners) checkCount(secure, quic, insecure int) error {
+	// same number of secure and quic listeners, and at least one
+	if secure == quic && secure > 0 {
+		// either no insecure listeners, or the same number as secure ones
+		if insecure == 0 || insecure == secure {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("inconsistent listeners count (secure:%v, quic:%v, insecure:%v)",
```

Comment:
In cases like this I would expect to get the why too, not only the what. Inconsistent listeners count... no many insecure listeners or insecure listeners cannot be in the same number as the secure ones.

Reply:
I'll improve this log entry merging `agent/httpserver` @karasz